src/Microsoft.Azure.WebJobs.Host/Bindings/ClonerContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    ///  Context passed to <see cref="AutoResolveAttribute"/> to aide in resolution. 
+    /// </summary>
+    internal class ClonerContext
+    {
+        public string MethodName { get; set; }
+
+        public INameResolver NameResolver { get; set; }
+
+        public static ClonerContext New(INameResolver resolver)
+        {
+            return new ClonerContext
+            {
+                NameResolver = resolver,
+            };
+        }
+
+        public static ClonerContext New(INameResolver resolver, ParameterInfo parameter)
```

Comment:
What happens in the case of Binder.BindAsync calls where the ParameterInfo will be fake - it still has a valid MemberInfo, right? Probably be good to do a Functions Node test to make sure